src/components/I18nProvider.js

```diff
@@ -0,0 +1,25 @@
+import { Component } from 'preact';
+
+import getTranslations from '../utils/i18n';
+
+export default class I18nProvider extends Component {
+    constructor(props) {
+        super(props);
+
+        const { language } = this.props;
+
+        this.state = { i18n: getTranslations(language) };
+    }
+
+    getChildContext() {
+        const { i18n } = this.state;
+
+        return { i18n };
+    }
+
+    render() {
+        const { children } = this.props;
+
+        return children[0];
```

Comment:
can't we just `return children;` ?

Reply:
React supports `children` as a single element or an array, In Preact `children` is always an array so to avoid having additional div and to return single element I just return the first one.

Reply:
Would it create additional div when you return an array?

Reply:
Preact components don't have a support for returning arrays